contracts/v1/CommitManagerV1New.sol

```diff
@@ -0,0 +1,424 @@
+            ProfileStorage ps = profileStorage;
+
+            revert ServiceAgreementErrorsV1.NodeNotInShardingTable(
+                args.rightNeighbourHoodEdge,
+                ps.getNodeId(args.rightNeighbourHoodEdge),
+                ps.getAsk(args.rightNeighbourHoodEdge),
+                stakingStorage.totalStakes(args.rightNeighbourHoodEdge)
+            );
+        }
+
+        ShardingTableStructs.Node memory closestNode = shardingTableStorage.getNode(args.closestNode);
+        ShardingTableStructs.Node memory leftNeighbourhoodEdge = shardingTableStorage.getNode(
+            args.leftNeighbourHoodEdge
+        );
+        ShardingTableStructs.Node memory rightNeighbourhoodEdge = shardingTableStorage.getNode(
+            args.rightNeighbourHoodEdge
+        );
+
+        bool isBetween = (leftNeighbourhoodEdge.index > rightNeighbourhoodEdge.index)
+            ? ((closestNode.index > leftNeighbourhoodEdge.index) || (closestNode.index < rightNeighbourhoodEdge.index))
+            : ((closestNode.index > leftNeighbourhoodEdge.index) && (closestNode.index < rightNeighbourhoodEdge.index));
+
+        if (!isBetween) {
+            revert CommitManagerErrorsV1.closestNodeNotInNeighbourhood(
+                agreementId,
+                args.leftNeighbourHoodEdge,
+                args.rightNeighbourHoodEdge,
+                args.closestNode,
+                args.epoch,
+                block.timestamp
+            );
+        }
+
+        uint256 numberOfNodes = shardingTableStorage.nodesCount();
+        uint256 clockwiseDistance = (rightNeighbourhoodEdge.index + numberOfNodes - leftNeighbourhoodEdge.index) %
+            numberOfNodes;
+        uint256 counterclockwiseDistance = (leftNeighbourhoodEdge.index +
+            numberOfNodes -
+            rightNeighbourhoodEdge.index) % numberOfNodes;
+
+        uint256 indexDistance = (clockwiseDistance < counterclockwiseDistance)
+            ? clockwiseDistance
+            : counterclockwiseDistance;
+
+        //distance between 20 nodes is 19 (this shold be constant)
+        if (!(indexDistance == 19)) {
+            revert CommitManagerErrorsV1.negihbourhoodWrongSize(
+                agreementId,
+                args.leftNeighbourHoodEdge,
+                args.rightNeighbourHoodEdge,
+                numberOfNodes,
+                20,
+                indexDistance,
+                args.epoch,
+                block.timestamp
+            );
+        }
+
+        uint256 hashRingNeighbourhoodDistance = calculateHashRingDistance(
+            leftNeighbourhoodEdge.hashRingPosition,
+            rightNeighbourhoodEdge.hashRingPosition
+        );
+
+        bytes32 keywordHash = hashingProxy.callHashFunction(args.hashFunctionId, args.keyword);
+        bytes32 nodeIdHash = hashingProxy.callHashFunction(args.hashFunctionId, profileStorage.getNodeId(identityId));
+
+        uint256 distance = calculateHashRingDistance(
```

Comment:
this is the same as `hashRingNeighbourhoodDistance`. Did you mean to use `keywordHash` and `nodeIdHash` as arguments ?